### ISSUE TITLE
Setting the nr_topics to 50 , but still getting 3000 topics. I have around 200000 documents. I apply umap, hdbscan, MaximalMarginalRelevance,ClassTfidfTransformer, CountVectorizer. Any reason to why it doesn't create only 50 topics?

### ISSUE THREAD

**Issue by VickyAnP** [MaartenGr/BERTopic#1441]

**Comment by MaartenGr** [comment]
Could you share your code, including the version of BERTopic? Make sure to be as complete as possible.

**Comment by VickyAnP** [comment]
Hi Maarten! Thank you for your prompt reply. I will paste down the code. The version i had before was bertopic==0.15.0 . Then i upgraded bertopic to see if anything changes and the version is still 0.15.0. And now when i ran the same script, i got only 30 topics instead of thousands, despite setting it to 50.  What do you think the issue would be?

MODEL = 'all-mpnet-base-v2'
sentence_model = SentenceTransformer(MODEL) 
umap_model_chosen = UMAP(n_neighbors=10, min_dist=0.05, metric='cosine',random_state=42)
hdbscan_model_chosen = HDBSCAN(min_cluster_size=10, metric='euclidean', gen_min_span_tree = True,
                    cluster_selection_method='eom', prediction_data=True, min_samples=5)
vectorizer_model = CountVectorizer(stop_words="english", lowercase=True, ngram_range=(1, 2), min_df=2) 
ctfidf_model_chosen = ClassTfidfTransformer(reduce_frequent_words=True)
representation_model_chosen = MaximalMarginalRelevance(diversity=0.3)
topic_model = BERTopic(umap_model = umap_model_chosen,
                      hdbscan_model = hdbscan_model_chosen,
                      vectorizer_model = vectorizer_model,
                      calculate_probabilities = False, 
                      nr_topics ="50",
                      language ="english",
                      ctfidf_model = ctfidf_model_chosen,
                      representation_model = representation_model_chosen,
                      top_n_words = 50,
                      min_topic_size = 100) 
topics_calc, probs = topic_model.fit_transform(abstracts_text, abstract_embeddings)

**Comment by VickyAnP** [comment]
Ahh, that is a silly mistake. Thank you for pointing out. Yes, now i get exactly the nr. of clusters mentioned in nr_topics. Really appreciate the fast response.

**Comment by MaartenGr** [comment]
No problem! Glad I could be of help.
